extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/handler/GenericThingHandler.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.mqtt.generic.handler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.binding.mqtt.generic.MqttBindingConstants;
+import org.eclipse.smarthome.binding.mqtt.generic.internal.AbstractMqttThingValue;
+import org.eclipse.smarthome.binding.mqtt.generic.internal.ColorValue;
+import org.eclipse.smarthome.binding.mqtt.generic.internal.MqttChannelTypeProvider;
+import org.eclipse.smarthome.binding.mqtt.generic.internal.NumberValue;
+import org.eclipse.smarthome.binding.mqtt.generic.internal.OnOffValue;
+import org.eclipse.smarthome.binding.mqtt.generic.internal.TextValue;
+import org.eclipse.smarthome.binding.mqtt.generic.internal.TransformationServiceProvider;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection;
+import org.eclipse.smarthome.io.transport.mqtt.MqttException;
+
+/**
+ * This handler manages manual created Things with manually added channels to link to MQTT topics.
+ *
+ * @author David Graeff - Initial contribution
+ */
+@NonNullByDefault
+public class GenericThingHandler extends AbstractThingHandler implements ChannelStateUpdateListener {
+    final Map<ChannelUID, ChannelState> channelDataByChannelUID = new HashMap<>();
+
+    public GenericThingHandler(Thing thing, MqttChannelTypeProvider provider,
+            @Nullable TransformationServiceProvider transformationServiceProvider) {
+        super(thing, provider, transformationServiceProvider);
```

Comment:
Hello, I have a quick question regarding this MqttChannelTypeProvider. I saw it is used in the HomieThingHandler but nowhere in the AbstractThingHandler and GenericThingHandler so why do you require it in the constructor. Sorry if I missed something just asking out of curiosity.